tests/unit/sample_plans/base.yaml

```diff
@@ -0,0 +1,155 @@
+plan: |
```

Comment:
the way I understand this PR is:
* given a cloud, there is only one overall correct upgrade plan
* the overall plan can be segmented into sub-plans (e.g. control plane, data plane, some set of empty hypervisors)
* we should ensure that we are able to generate the correct overall plan for as many cloud models as possible
* our ability to segment the plan into sub-plans, or exclude units that cannot be upgraded for some reason, should be tested separately and independently from the input model

If the above is what we're trying to do, I think sticking to only testing the overall control plane + data plane (assuming all hypervisors are empty) upgrade plan is correct.

Reply:
In all these test I'm focused only on testing upgrade plan base on deployment and their configuration, so I did not try to do this tests for any sanity check or filtering not empty hypervisors or for testing different user inputs.

We can define more parts in the yaml, like CLIargs, number of VMs, etc. But those parts should be tested by unittests itself, since here we are focusing on real world deployment with lot of applications e.g. huge yaml file with all application definition.